src/main/java/cam72cam/immersiverailroading/blocks/BlockRailBase.java

```diff
@@ -294,6 +293,13 @@ public boolean onBlockActivated(World worldIn, BlockPos pos, IBlockState state,
 		Block block = Block.getBlockFromItem(stack.getItem());
 		TileRailBase te = TileRailBase.get(worldIn, pos);
 		if (te != null) {
+			if (stack.getItem() == IRItems.ITEM_SWITCH_KEY) {
+				if (!worldIn.isRemote) {
+					te.cycleSwitchForced();
```

Comment:
Why not return the new state here so you don't need to look it up down below?

Reply:
Great question. The answer is because I created the method before I knew I needed to retrieve the new state again. Just so I know we're on the same page, you mean have cycleSwitchForced() return the new state to use in the chat message calls?

Reply:
Correct